components/chart/index.tsx

```diff
@@ -55,8 +59,35 @@ function Chart({
     ZOOM_OUT: () => changeZoomLevel(10),
   };
 
+  const toggleDrawer = () => {
+    setDrawerOpen((prev) => !prev);
+  };
+  const drawerRef = useRef(null);
```

Comment:
let's move this to the beginning, before defining the functions

Reply:
done